src/Umbraco.Web/Models/Link.cs

```diff
@@ -0,0 +1,13 @@
+﻿using Umbraco.Core;
+
+namespace Umbraco.Web.Models
+{
+    public class Link
```

Comment:
Should there be a `IPublishedContent` property that exposes the content/media (e.g. `public IPublishedContent Content { get; set; }`)? The `MultiUrlPickerPropertyConverter` already does the lookup, so it could easily set this property.

And doesn't the `Udi` already contain what type of entity it is (`umb://document/{guid}`)? That way, the both `Udi` and `Content` properties could be used to get the link type.